core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -748,6 +759,12 @@ private[spark] class ExternalSorter[K, V, C](
 
     context.taskMetrics.memoryBytesSpilled += memoryBytesSpilled
     context.taskMetrics.diskBytesSpilled += diskBytesSpilled
+    context.taskMetrics.shuffleWriteMetrics.filter(_ => bypassMergeSort).foreach { m =>
```

Comment:
I was reading through `ExternalSorter` to try to understand how shuffle write time metrics are calculated and came across this line.  This style is confusing to a casual reader: it looks like the logic here is "if shuffle write metrics are defined and merge sort is bypassed, then run this block", but this is slightly obfuscated by the fact that we're filtering an option with a filter function that doesn't depend on that option's value.

For next time, I think we should just use a simple `if` statement instead.